client/reader/user-stream/index.tsx

```diff
@@ -0,0 +1,92 @@
+import page from '@automattic/calypso-router';
+import { useTranslate } from 'i18n-calypso';
+import SectionNav from 'calypso/components/section-nav';
+import NavItem from 'calypso/components/section-nav/item';
+import NavTabs from 'calypso/components/section-nav/tabs';
+import UserComments from './views/comments';
+import UserLikes from './views/likes';
+import UserLists from './views/lists';
+import UserPosts from './views/posts';
+import UserReposts from './views/reposts';
+import './style.scss';
+
+interface NavigationItem {
+	label: string;
+	path: string;
+	selected: boolean;
+}
+
+interface UserStreamProps {
+	streamKey?: string;
+	userId: string;
+}
+
+const UserStream = ( { streamKey, userId }: UserStreamProps ) => {
+	const translate = useTranslate();
+	const currentPath = page.current;
+
+	const navigationItems: NavigationItem[] = [
+		{
+			label: translate( 'Posts' ),
+			path: `/read/users/${ userId }`,
+			selected: currentPath === `/read/users/${ userId }`,
+		},
+		{
+			label: translate( 'Comments' ),
+			path: `/read/users/${ userId }/comments`,
+			selected: currentPath === `/read/users/${ userId }/comments`,
+		},
+		{
+			label: translate( 'Likes' ),
+			path: `/read/users/${ userId }/likes`,
+			selected: currentPath === `/read/users/${ userId }/likes`,
+		},
+		{
+			label: translate( 'Reposts' ),
+			path: `/read/users/${ userId }/reposts`,
+			selected: currentPath === `/read/users/${ userId }/reposts`,
+		},
+		{
+			label: translate( 'Lists' ),
+			path: `/read/users/${ userId }/lists`,
+			selected: currentPath === `/read/users/${ userId }/lists`,
+		},
+	];
+
+	const selectedTab = navigationItems.find( ( item ) => item.selected )?.label || '';
+
+	const renderContent = (): React.ReactNode => {
+		const basePath = currentPath.split( '?' )[ 0 ];
+
+		switch ( basePath ) {
```

Comment:
One issue I see here is what happens when none of these routes match? Testing it it looks like we just get a blank page.

Reply:
That's true. A 404 would be better. But I think serving a blank page instead of 404 may be pervasive in the reader in production. For example, if I go to https://wordpress.com/read/lsdlfkjklas, I get a blank page. So, this might be worthy of a larger follow-up?